platform/linuxbsd/freedesktop_portal_desktop.cpp

```diff
@@ -126,6 +130,317 @@ uint32_t FreeDesktopPortalDesktop::get_appearance_color_scheme() {
+	DBusMessageIter dict_iter;
+	DBusMessageIter var_iter;
+	dbus_message_iter_open_container(p_iter, DBUS_TYPE_DICT_ENTRY, nullptr, &dict_iter);
+	append_dbus_string(&dict_iter, p_key);
+
+	if (p_as_byte_array) {
+		DBusMessageIter arr_iter;
+		dbus_message_iter_open_container(&dict_iter, DBUS_TYPE_VARIANT, "ay", &var_iter);
+		dbus_message_iter_open_container(&var_iter, DBUS_TYPE_ARRAY, "y", &arr_iter);
+		CharString cs = p_value.utf8();
+		const char *cs_ptr = cs.get_data();
+		do {
+			dbus_message_iter_append_basic(&arr_iter, DBUS_TYPE_BYTE, cs_ptr);
+		} while (*cs_ptr++);
+		dbus_message_iter_close_container(&var_iter, &arr_iter);
+	} else {
+		dbus_message_iter_open_container(&dict_iter, DBUS_TYPE_VARIANT, "s", &var_iter);
+		append_dbus_string(&var_iter, p_value);
+	}
+
+	dbus_message_iter_close_container(&dict_iter, &var_iter);
+	dbus_message_iter_close_container(p_iter, &dict_iter);
+}
+
+void FreeDesktopPortalDesktop::append_dbus_dict_bool(DBusMessageIter *p_iter, const String &p_key, bool p_value) {
+	DBusMessageIter dict_iter;
+	DBusMessageIter var_iter;
+	dbus_message_iter_open_container(p_iter, DBUS_TYPE_DICT_ENTRY, nullptr, &dict_iter);
+	append_dbus_string(&dict_iter, p_key);
+
+	dbus_message_iter_open_container(&dict_iter, DBUS_TYPE_VARIANT, "b", &var_iter);
+	{
+		int val = p_value;
+		dbus_message_iter_append_basic(&var_iter, DBUS_TYPE_BOOLEAN, &val);
+	}
+
+	dbus_message_iter_close_container(&dict_iter, &var_iter);
+	dbus_message_iter_close_container(p_iter, &dict_iter);
+}
+
+bool FreeDesktopPortalDesktop::file_chooser_parse_response(DBusMessageIter *p_iter, bool &r_cancel, Vector<String> &r_urls) {
+	ERR_FAIL_COND_V(dbus_message_iter_get_arg_type(p_iter) != DBUS_TYPE_UINT32, false);
+
+	dbus_uint32_t resp_code;
+	dbus_message_iter_get_basic(p_iter, &resp_code);
+	if (resp_code != 0) {
+		r_cancel = true;
+	} else {
+		r_cancel = false;
+		ERR_FAIL_COND_V(!dbus_message_iter_next(p_iter), false);
+		ERR_FAIL_COND_V(dbus_message_iter_get_arg_type(p_iter) != DBUS_TYPE_ARRAY, false);
+
+		DBusMessageIter dict_iter;
+		dbus_message_iter_recurse(p_iter, &dict_iter);
+		while (dbus_message_iter_get_arg_type(&dict_iter) == DBUS_TYPE_DICT_ENTRY) {
+			DBusMessageIter iter;
+			dbus_message_iter_recurse(&dict_iter, &iter);
+			if (dbus_message_iter_get_arg_type(&iter) == DBUS_TYPE_STRING) {
+				const char *key;
+				dbus_message_iter_get_basic(&iter, &key);
+				dbus_message_iter_next(&iter);
+
+				DBusMessageIter var_iter;
+				dbus_message_iter_recurse(&iter, &var_iter);
+				if (strcmp(key, "uris") == 0) {
+					if (dbus_message_iter_get_arg_type(&var_iter) == DBUS_TYPE_ARRAY) {
+						DBusMessageIter uri_iter;
+						dbus_message_iter_recurse(&var_iter, &uri_iter);
+						while (dbus_message_iter_get_arg_type(&uri_iter) == DBUS_TYPE_STRING) {
+							const char *value;
+							dbus_message_iter_get_basic(&uri_iter, &value);
+							r_urls.push_back(String::utf8(value).trim_prefix("file://"));
```

Comment:
```suggestion
							r_urls.push_back(String::utf8(value).trim_prefix("file://").uri_decode());
```

e.g. Convert `%20` into whitespaces.